ultra/tests/test_scenarios.py

```diff
@@ -125,6 +125,106 @@ def test_interface_generate_multiagent(self):
             print(err)
             self.assertTrue(False)
 
+    def test_interface_generate_with_shuffled_missions(self):
+        TASK = "00-multiagent"
+        LEVEL = "shuffle_test"
+        ROOT_DIR = "tests/scenarios"
+        SAVE_DIR = os.path.join(
+            ScenariosTest.OUTPUT_DIRECTORY, "maps/shuffle_test-with-shuffle/"
+        )
+
+        if os.path.exists(SAVE_DIR):
+            shutil.rmtree(SAVE_DIR)
+
+        generate_command = (
+            "python ultra/scenarios/interface.py generate "
+            f"--task {TASK} "
+            f"--level {LEVEL} "
+            f"--root-dir {ROOT_DIR} "
+            f"--save-dir {SAVE_DIR}"
+        )
+
+        # Test generation without the --no-shuffle flag.
+        try:
+            os.system(generate_command)
+        except Exception as error:
+            print(error)
+            self.assertTrue(False)
+
+        first_missions = None
+        num_scenario_missions_that_differ = 0
+
+        for dirpath, _, files in os.walk(SAVE_DIR):
+            if "missions.pkl" in files and "train" in dirpath:
```

Comment:
I may think adding a comment which states that the test scenarios only will contain 1 mission per scenario thus shuffling will take no affect on the agent's mission, where as train scenarios can have multiple missions thus shuffling will alter agents' missions. I do think the comment is a "must have" but can provide a gentle reminder